haskellPackages.readline: fix Setup.hs to work with Cabal 3

samuelrivas opened the pull request:
See https://github.com/NixOS/nixpkgs/pull/111985/ for previous discussion

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

readline is broken for Cabal3 build. I fixed it previously in this [pull request](https://github.com/NixOS/nixpkgs/pull/111985/files), but the fix seems to have disappeared. I just reapplied it.

Is there a better way to keep this fix in place for the future? As you can seen in the previous PR, we tried to reach the maintainers of this package but we didn't succeed.


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

sternenseemann committed:
skawarePackages.buildManPages: derivation wrapper for skaware man page ports

GitHub user flexibeast has been porting the html documentation from
skarnet.org to mdoc, making them available as man pages. While the
documentation is non authorative, it is certainly useful and is also
linked from skarnet.org.

buildManPages implements the common mkDerivation machinery common to all
ported man page packages / repositories.

sternenseemann committed:
execline-man-pages: init at 2.8.0.1.1

sternenseemann committed:
s6-man-pages: init at 2.10.0.3.1

sternenseemann committed:
s6-networking-man-pages: init at 2.4.1.1.1

Commit:
neovide: fix meta.mainProgram

it was trying to start neovide-unstable

thoughtpolice committed:
nextpnr: 2021.07.28 -> 2021.08.06

Signed-off-by: Austin Seipp <aseipp@pobox.com>

Commit:
cargo-watch: fix build on darwin

samuelrivas committed:
haskellPackages.readline: fix Setup.hs to work with Cabal 3

See https://github.com/NixOS/nixpkgs/pull/111985/ for previous discussion

primeos committed:
llvmPackages_git: 2021-07-16 -> 2021-08-03

primeos committed:
Merge pull request #133929 from r-ryantm/auto-update/wob

wob: 0.11 -> 0.12

primeos committed:
Merge pull request #133935 from primeos/chromiumDev

chromiumDev: Fix the installation phase

primeos committed:
Merge pull request #133938 from primeos/signal-desktop

signal-desktop: 5.13.0 -> 5.13.1

mweinelt committed:
Merge pull request #133791 from r-ryantm/auto-update/python3.8-celery

python38Packages.celery: 5.1.1 -> 5.1.2

lukegb committed:
Merge pull request #133919 from Kranzes/auto-update/python3.8-hg-evolve

python38Packages.hg-evolve: 10.3.2 -> 10.3.3

andir committed:
sconsPackages: expose the python version used with scons

This allows users of scons to pick the correct version of python.
Previously we had issues with some build systems not picking the right
python3 version when adding additional python modules to the build
environment. A famous example of this is mongodb where additional python
modules are required to run the scons build.

This is change doesn't introduce rebuilds (to the best of my knowledge)
as it only adds a passthru argument and changes how we pass the python
version around.

andir committed:
mongodb: pick python version from scons

The python modules that we to the enviroment should always come from the
same python version as scons as otherwise they can't be loaded.

This unbreaks the build of mongodb 4.2 which pinned python to 3.8 while
our python attribute moved on to 3.9 which is what scons uses now.

q60 committed:
maintainers: change github username

Kranzes committed:
intel-media-sdk: 21.2.3 -> 21.3.1

sternenseemann committed:
fcft: 2.4.4 -> 2.4.5

https://codeberg.org/dnkl/fcft/releases/tag/2.4.5

r-ryantm committed:
libite: 2.2.0 -> 2.4.0

aanderse committed:
redmine: 4.2.1 -> 4.2.2

r-ryantm committed:
libpg_query: 13-2.0.5 -> 13-2.0.7

r-ryantm committed:
libqalculate: 3.19.0 -> 3.20.1

r-ryantm committed:
yamllint: 1.26.1 -> 1.26.2

aanderse committed:
Merge pull request #133671 from jlesquembre/caddy

nixos/caddy: update ca option

jtojnar committed:
Merge pull request #133656 from jakobrs/patch-2

cheese: add pipewire as a dependency

lukegb committed:
Merge pull request #133009 from yorickvP/asterisk-upd

asterisk: 13.38.2 -> 13.38.3, 16.17.0 -> 16.19.1, 17.9.3 -> 17.9.4, 18.3.0 -> 18.5.1

r-ryantm committed:
zlib-ng: 2.0.2 -> 2.0.5

fabaff committed:
Merge pull request #133993 from q60/patch-1

maintainers: change github username

fabaff committed:
Merge pull request #133917 from SFrijters/python-wrapio-2.0.0

python3Packages.wrapio: 1.0.0 -> 2.0.0

AndersonTorres committed:
Merge pull request #109302 from SrTobi-Forks/tobias/grub-savedefault

nixos/grub: implements GRUB_SAVEDEFAULT feature

fabaff committed:
python3Packages.cachelib: 0.2.0 -> 0.3.0

risicle committed:
Merge pull request #133718 from r-ryantm/auto-update/python3.8-authlib

python38Packages.authlib: 0.15.3 -> 0.15.4

r-ryantm committed:
libxlsxwriter: 1.1.1 -> 1.1.3

r-ryantm committed:
totem-pl-parser: 3.26.5 -> 3.26.6

r-ryantm committed:
lifecycled: 3.1.0 -> 3.2.0

r-ryantm committed:
lightburn: 0.9.23 -> 1.0.00

r-ryantm committed:
verilator: 4.202 -> 4.210

r-ryantm committed:
vips: 8.10.6 -> 8.11.2

doronbehar committed:
Merge pull request #133934 from r-ryantm/auto-update/zoom

Kranzes committed:
klayout: 0.26.10 -> 0.27.3

r-ryantm committed:
maxcso: 1.12.0 -> 1.13.0

r-ryantm committed:
microplane: 0.0.32 -> 0.0.33

r-ryantm committed:
mob: 1.4.0 -> 1.8.0

r-ryantm committed:
mockgen: 1.5.0 -> 1.6.0

figsoda committed:
vimPlugins: fix update script indentation

r-ryantm committed:
toybox: 0.8.4 -> 0.8.5

Artturin committed:
Merge pull request #134006 from Kranzes/auto-update/intel-media-sdk

r-ryantm committed:
gnome.totem: 3.38.0 -> 3.38.1

Artturin committed:
Merge pull request #133902 from Kranzes/auto-update/gallery_dl

r-ryantm committed:
ulauncher: 5.9.0 -> 5.11.0

Artturin committed:
Merge pull request #133973 from Kranzes/auto-update/ibus-table

ibus-engines.table: 1.12.4 -> 1.14.0

fabaff committed:
Merge pull request #133880 from r-ryantm/auto-update/disfetch

disfetch: 2.9 -> 2.10

Artturin committed:
Merge pull request #133940 from Kranzes/auto-update/homebank

homebank: 5.5.2 -> 5.5.3

romildo committed:
marwaita: 10.2 -> 10.3

Artturin committed:
Merge pull request #133936 from Kranzes/auto-update/grisbi

Artturin committed:
Merge pull request #133927 from Kranzes/auto-update/half

half: 2.1.0 -> 2.2.0

Artturin committed:
Merge pull request #133918 from Kranzes/auto-update/fwup

fwup: 1.8.4 -> 1.9.0

Artturin committed:
Merge pull request #133903 from Kranzes/auto-update/flent

flent: 1.3.2 -> 2.0.1

AndersonTorres committed:
cbftp: init at 1173

Artturin committed:
Merge pull request #133868 from Kranzes/auto-update/buf

buf: 0.49.0 -> 0.50.0

Artturin committed:
Merge pull request #134056 from r-ryantm/auto-update/ulauncher

ulauncher: 5.9.0 -> 5.11.0

fabaff committed:
Merge pull request #133578 from smancill/python-chess

python3Packages.chess: init at 1.6.1

Artturin committed:
Merge pull request #133881 from Kranzes/auto-update/dolt

dolt: 0.27.2 -> 0.27.3

Artturin committed:
Merge pull request #133910 from r-ryantm/auto-update/xlog

xlog: 2.0.20 -> 2.0.23

Artturin committed:
Merge pull request #134055 from r-ryantm/auto-update/totem

gnome.totem: 3.38.0 -> 3.38.1

Artturin committed:
Merge pull request #133878 from Kranzes/auto-update/eksctl

eksctl: 0.60.0 -> 0.61.0

Artturin committed:
Merge pull request #133853 from Artturin/bigloo

fabaff committed:
python3Packages.total-connect-client: 0.58 -> 2021.7.1

lovesegfault committed:
Merge pull request #134050 from figsoda/fix-vim-update

vimPlugins: fix update script indentation

Artturin committed:
Merge pull request #134027 from r-ryantm/auto-update/lifecycled

lifecycled: 3.1.0 -> 3.2.0

figsoda committed:
vimPlugins: update

figsoda committed:
vimPlugins.nvim-expand-expr: init at 2021-08-14

Artturin committed:
Merge pull request #133999 from r-ryantm/auto-update/libite

libite: 2.2.0 -> 2.4.0

ryantm committed:
Merge pull request #134047 from r-ryantm/auto-update/mockgen

mockgen: 1.5.0 -> 1.6.0

ryantm committed:
Merge pull request #134029 from r-ryantm/auto-update/lightburn

lightburn: 0.9.23 -> 1.0.00

ryantm committed:
Merge pull request #134046 from r-ryantm/auto-update/mob

mob: 1.4.0 -> 1.8.0

ryantm committed:
Merge pull request #134016 from r-ryantm/auto-update/zlib-ng

zlib-ng: 2.0.2 -> 2.0.5

ryantm committed:
Merge pull request #133924 from r-ryantm/auto-update/xrandr-invert-colors

xrandr-invert-colors: 0.01 -> 0.02

ryantm committed:
Merge pull request #133859 from superherointj/package-linode-cli-5.6.0

linode-cli: 5.5.2 -> 5.6.0

laMudri committed:
vscode: update example to match #131589

vic committed:
Fix import path.

Trying to create a simple flake with input:

```
{
     inputs.nixpkgs-lib.url = "github:NixOS/nixpkgs?dir=lib";
     outputs = {nixpkgs-lib, ...}: { };
}
```

Fails when trying to evaluate the nixpkgs-lib flake with an error like:

```
error: getting status of '/nix/store/xxxx-source/lib/lib': No such file or directory
(use '--show-trace' to show detailed location information)
```

roberth committed:
Merge pull request #134068 from laMudri/bbenoist-Nix

vscode: update example to match #131589

primeos committed:
Merge pull request #134014 from primeos/llvmPackages_git

llvmPackages_git: 2021-07-16 -> 2021-08-03

roberth committed:
Merge pull request #133731 from Nervengift/patch-1

Eclipse: add libsecret dependency

Commit:
pass2csv: init at 0.3.1

Commit:
python38Packages.pytwitchapi: init at 2.3.2

lukegb committed:
dockerTools.pullImage: fix for skopeo 1.4.x

skopeo 1.4.x doesn't accept --src-tls-verify as a flag to the *program*,
only as a flag to copy; we must pass it after the "copy" verb, or it
will fail with:

> FATA[0000] unknown flag: --src-tls-verify

roberth committed:
Merge pull request #134071 from lukegb/skopeo-cli-change

dockerTools.pullImage: fix for skopeo 1.4.x

Profpatsch committed:
rabbitmq: Remove Profpatsch as maintainer

I have effectively not really maintained it in a while, since I had no
use-case.

Now that ryantm-bot has stopped reacting to updates of the package, I
don’t see this improving.

I hope somebody else steps up, an up-to-date rabbitmq in nixpkgs is
quite important.

Kranzes committed:
webdis: 0.1.15 -> 0.1.16

SuperSandro2000 committed:
mlt: cleanup

ryantm committed:
Merge pull request #134042 from r-ryantm/auto-update/microplane

microplane: 0.0.32 -> 0.0.33

SuperSandro2000 committed:
Merge pull request #133942 from sternenseemann/catgirl-misc-fixes

fabaff committed:
python3Packages.howdoi: 2.0.16 -> 2.0.17

fabaff committed:
python3Packages.howdoi: enable tests

SuperSandro2000 committed:
Merge pull request #133920 from raboof/zd1211-firmware-less-fixed-output-derivation

zd1211-firmware: less fixed-output derivation

SuperSandro2000 committed:
Merge pull request #134075 from Profpatsch/rabbitmq-maintainer

rabbitmq: Remove Profpatsch as maintainer

SuperSandro2000 committed:
Merge pull request #134076 from Kranzes/auto-update/webdis

webdis: 0.1.15 -> 0.1.16

SuperSandro2000 committed:
Merge pull request #134077 from SuperSandro2000/mlt

mlt: cleanup

SuperSandro2000 committed:
Merge pull request #133824 from WolfangAukang/pass2csv-init

pass2csv: init at 0.3.1

SuperSandro2000 committed:
Merge pull request #133871 from WolfangAukang/pyTwitchAPI-init

python38Packages.pytwitchapi: init at 2.3.2

SuperSandro2000 committed:
Merge pull request #133844 from figsoda/nvim-expand-expr

fabaff committed:
Merge pull request #133922 from fabaff/bump-ipinfo

ipinfo: 2.0.1 -> 2.0.2

SuperSandro2000 committed:
Merge pull request #133857 from r-ryantm/auto-update/aws-c-common

aws-c-common: 0.6.8 -> 0.6.9

SuperSandro2000 committed:
Merge pull request #134044 from fabaff/bump-howdoi

SuperSandro2000 committed:
Merge pull request #134060 from romildo/upd.marwaita

marwaita: 10.2 -> 10.3

SuperSandro2000 committed:
homebank: switch to pname + version

SuperSandro2000 committed:
dolt: format

r-burns committed:
audiowaveform: 1.4.2 -> 1.5.1, fix gtest subproject

Will be needed to fix build with gtest v1.11.0
(currently in staging-next #131928).
See https://github.com/bbc/audiowaveform/commit/7243739cb138e3952884786ba42694dd9ddc2c09

kini committed:
acl2: 8.3 -> 8.4

SuperSandro2000 committed:
Merge pull request #134015 from fabaff/bump-cachelib

python3Packages.cachelib: 0.2.0 -> 0.3.0

SuperSandro2000 committed:
Merge pull request #133842 from fabaff/bump-zeroconf

python3Packages.zeroconf: 0.34.3 -> 0.35.0

SuperSandro2000 committed:
Merge pull request #133843 from fabaff/bump-python-miio

python3Packages.python-miio: 0.5.6 -> 0.5.7

github-actions[bot] committed:
Merge master into haskell-updates

Artturin committed:
Merge pull request #134036 from Kranzes/auto-update/klayout

r-ryantm committed:
khal: 0.10.3 -> 0.10.4

SuperSandro2000 committed:
Merge pull request #134001 from aanderse/redmine

redmine: 4.2.1 -> 4.2.2

SuperSandro2000 committed:
Merge pull request #134078 from SuperSandro2000/dolt

dolt: format

SuperSandro2000 committed:
Merge pull request #134079 from SuperSandro2000/homebank

SuperSandro2000 committed:
Merge pull request #133997 from sternenseemann/fcft-2.4.5

fcft: 2.4.4 -> 2.4.5

r-ryantm committed:
agenda: 1.1.1 -> 1.1.2

SuperSandro2000 committed:
Merge pull request #133838 from fabaff/bump-pyhomematic

python3Packages.pyhomematic: 0.1.73 -> 0.1.74

SuperSandro2000 committed:
Merge pull request #133898 from r-ryantm/auto-update/writefreely

writefreely: 0.12.0 -> 0.13.1

leungbk committed:
vimPlugins: update

r-ryantm committed:
urh: 2.9.1 -> 2.9.2

SuperSandro2000 committed:
Merge pull request #133858 from superherointj/package-temporal-1.11.3

temporal: 1.11.2 -> 1.11.3

SuperSandro2000 committed:
Merge pull request #133856 from superherointj/package-linkerd-edge-21.8.2

linkerd_edge: 21.7.4 -> 21.8.2

leungbk committed:
vimPlugins.clever-f-vim: init at 2021-07-07

SuperSandro2000 committed:
Merge pull request #133864 from samuela/samuela/libressl_3_1

libressl: remove vulnerable libressl_3_1

r-ryantm committed:
glances: 3.2.2 -> 3.2.3

figsoda committed:
vimPlugins.plenary-nvim: don't build from luaPackages.plenary-nvim

Artturin committed:
Merge pull request #134010 from r-ryantm/auto-update/yamllint

davidak committed:
Merge pull request #134086 from r-ryantm/auto-update/agenda

agenda: 1.1.1 -> 1.1.2

davidak committed:
Merge pull request #134053 from r-ryantm/auto-update/toybox

toybox: 0.8.4 -> 0.8.5

davidak committed:
Merge pull request #133874 from Kranzes/auto-update/burpsuite

burpsuite: 2021.8 -> 2021.8.1

Artturin committed:
Merge pull request #133894 from r-ryantm/auto-update/xtermcontrol

xtermcontrol: 3.7 -> 3.8

Artturin committed:
Merge pull request #133863 from r-ryantm/auto-update/bcompare

r-ryantm committed:
erlang-ls: 0.17.0 -> 0.18.0

thoughtpolice committed:
yosys: 0.9+4221 -> 0.9+4272

Signed-off-by: Austin Seipp <aseipp@pobox.com>

r-burns committed:
libgpgerror: fix cross-compilation

Fixes e.g. pkgsCross.aarch64-multiplatform.libgpgerror

Mic92 committed:
Merge pull request #133588 from zhaofengli/nuget-to-nix

nuget-to-nix: init

kini committed:
sbcl: backport ARM segfault fix for large cores

tekeri committed:
rust: fix build of pam-sys package

r-ryantm committed:
python38Packages.mautrix: 0.10.2 -> 0.10.3

marsam committed:
Merge pull request #134003 from r-ryantm/auto-update/libpg_query

libpg_query: 13-2.0.5 -> 13-2.0.7

r-ryantm committed:
lethe: 0.5.1 -> 0.6.0

marsam committed:
catdoc: fix build on darwin

cdepillabout commented:
> Is there a better way to keep this fix in place for the future?

It might be a better idea to send this PR to the `haskell-updates` branch, and then back-port this PR after it is merged in there.

Each nixos release is branched-off from the `master` branch here in nixpkgs.  https://github.com/NixOS/nixpkgs/pull/111985 was sent to `release-20.09`, so it never ended up in `master`.  Sorry for not let telling you about this at the time.

If you send a PR to `haskell-updates`, then it should get merged into `master`.

I think there is some sort of action that lets us automatically create a backport, so hopefully it is not too much additional work for you.

> As you can seen in the previous PR, we tried to reach the maintainers of this package but we didn't succeed.

One thing I was wondering, is there a repo for this readline package on github?  If so, can you create an issue or send a PR there and link to it from a comment in the code?  If you are able to send a PR, then you should also just be able to `fetchpatch` from that PR, which would mean that we wouldn't need to carry this patch around in nixpkgs.

If there is no upstream repo, the we'll probably have to carry around this patch in nixpkgs.

fabaff committed:
python3Packages.python-box: 5.3.0 -> 5.4.0

fabaff committed:
python3Packages.icmplib: 3.0.0 -> 3.0.1

fabaff committed:
python3Packages.identify: 2.2.11 -> 2.2.13

fabaff committed:
python3Packages.pyvicare: 2.5.2 -> 2.6

r-ryantm committed:
python38Packages.gensim: 4.0.0 -> 4.0.1

happysalada committed:
vscode-extensions.elixir-lsp.vscode-elixir-ls: 0.7.0 -> 0.8.0

Mic92 committed:
Merge pull request #134119 from tekeri/rust-crate-override-pamsys

rust: fix build of pam-sys package

Mic92 committed:
Merge pull request #134109 from r-burns/libgpgerror-cross

libgpgerror: fix cross-compilation

roberth committed:
Merge pull request #134069 from vic/patch-1

Fix import path on `lib/flake.nix`

jtojnar committed:
bpb: init at unstable-2018-07-27

https://without.boats/blog/signing-commits-without-gpg/

SuperSandro2000 committed:
Merge pull request #134092 from leungbk/clever-f-vim

SuperSandro2000 committed:
unipicker: cleanup

SuperSandro2000 committed:
Merge pull request #133366 from rnpgp/rnp

rnp: 0.15.1 -> 0.15.2

SuperSandro2000 committed:
Merge pull request #133861 from Kranzes/auto-update/aws-c-io

aws-c-io: 0.9.1 -> 0.10.5

pasqui23 committed:
skrooge: 2.24.6 ->2.26.1 (#134048)

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

SuperSandro2000 committed:
Merge pull request #134090 from r-ryantm/auto-update/urh

SuperSandro2000 committed:
Merge pull request #134153 from SuperSandro2000/unipicker

unipicker: cleanup

SuperSandro2000 committed:
Merge pull request #133676 from nagy/spdx-3-14

spdx-license-list-data: 3.13 -> 3.14

zimbatm committed:
treefmt: 0.2.3 -> 0.2.5

Sorixelle committed:
nixos/nginx: fix typo in listenAdresses

SuperSandro2000 committed:
Merge pull request #134065 from fabaff/bump-total-connect-client

python3Packages.total-connect-client: 0.58 -> 2021.7.1

SuperSandro2000 committed:
Merge pull request #133888 from Kranzes/auto-update/erfa

liberfa: 1.7.1 -> 2.0.0

SuperSandro2000 committed:
Merge pull request #134082 from r-ryantm/auto-update/khal

khal: 0.10.3 -> 0.10.4

SuperSandro2000 committed:
Merge pull request #134148 from jtojnar/bpb

bpb: init at unstable-2018-07-27

SuperSandro2000 committed:
Merge pull request #134158 from NixOS/treefmt-0.2.5

treefmt: 0.2.3 -> 0.2.5

SuperSandro2000 committed:
Merge pull request #134161 from Sorixelle/fix/nginx-module-typo

nixos/nginx: fix typo in listenAdresses

SuperSandro2000 committed:
Merge pull request #134131 from fabaff/bump-icmplib

python3Packages.icmplib: 3.0.0 -> 3.0.1

oxalica committed:
vim_configurable: handle python3Dependencies

SuperSandro2000 committed:
Merge pull request #134012 from r-ryantm/auto-update/libqalculate

libqalculate: 3.19.0 -> 3.20.1

r-ryantm committed:
neo-cowsay: 1.0.1 -> 1.0.3

r-ryantm committed:
nextdns: 1.11.0 -> 1.35.0

SuperSandro2000 committed:
Merge pull request #134130 from fabaff/bump-pyvicare

python3Packages.pyvicare: 2.5.2 -> 2.6

SuperSandro2000 committed:
Merge pull request #134135 from fabaff/bump-python-box

python3Packages.python-box: 5.3.0 -> 5.4.0

fabaff committed:
Merge pull request #134183 from r-ryantm/auto-update/neo-cowsay

neo-cowsay: 1.0.1 -> 1.0.3

fabaff committed:
Merge pull request #134134 from fabaff/bump-identify

python3Packages.identify: 2.2.11 -> 2.2.13

primeos committed:
Merge pull request #134094 from r-ryantm/auto-update/glances

glances: 3.2.2 -> 3.2.3

gebner committed:
Merge pull request #133884 from r-ryantm/auto-update/embree

embree: 3.13.0 -> 3.13.1

r-ryantm committed:
nzbget: 21.0 -> 21.1

gebner committed:
sumneko-lua-language-server: fix tests on aarch64

romildo committed:
Merge pull request #133879 from r-ryantm/auto-update/xfdashboard

xfce.xfdashboard: 0.9.2 -> 0.9.3

TredwellGit committed:
linux: 4.4.280 -> 4.4.281

TredwellGit committed:
linux: 4.9.279 -> 4.9.280

TredwellGit committed:
linux: 5.13.10 -> 5.13.11

Kranzes committed:
php74Extensions.couchbase: 3.1.2 -> 3.2.0

r-ryantm committed:
odpic: 4.1.0 -> 4.2.1

ajs124 committed:
wimboot: 2.6.0 -> 2.7.3

Kranzes committed:
vultr-cli: 2.4.1 -> 2.7.0

Kranzes committed:
mtools: 4.0.34 -> 4.0.35

SuperSandro2000 committed:
wezterm: 20210502-154244-3f7122cb -> 20210814-124438-54e29167

SuperSandro2000 committed:
Merge pull request #128781 from PrivateStorageio/skip-python-module-parsedatetime-tests

pythonPackages.parsedatetime: skip impure tests

andir committed:
Merge pull request #133921 from r-ryantm/auto-update/libyaml-cpp

libyamlcpp: 0.6.3 -> 0.7.0

andir committed:
Merge pull request #133873 from r-ryantm/auto-update/wimlib

wimlib: 1.13.2 -> 1.13.4

oxalica committed:
vimPlugins: update

oxalica committed:
vimPlugins.fcitx-vim: init at 2021-08-15

gebner committed:
Merge pull request #130326 from mjlbach/bump/sumneko-2.2.3

sumneko-lua-language-server: 1.20.2 -> 2.2.3

gebner committed:
sumneko-lua-language-server: 2.2.3 -> 2.3.6

r-ryantm committed:
oneDNN: 2.2.1 -> 2.3.2

r-ryantm committed:
onedrive: 2.4.12 -> 2.4.13

ajs124 committed:
Merge pull request #134206 from helsinki-systems/upd/wimboot

wimboot: 2.6.0 -> 2.7.3

r-ryantm committed:
oneshot: 1.4.1 -> 1.5.0

roberth committed:
treewide: runCommandNoCC -> runCommand

This has been synonymous for ~5y.

roberth committed:
yarn2nix-moretea: Generate runCommand instead of -NoCC

This has been synonymous for ~5y.

roberth committed:
treewide: runCommandNoCC -> runCommand in generated code

This has been synonymous for ~5y.

Note that many of these runCommand bindings are unused, but that's
ok for generated code.

roberth committed:
runCommandNoCC: deprecate

It has been synonymous with runCommand ~5y.

roberth committed:
treewide: runCommandNoCCLocal -> runCommandLocal

It has always been a synonym.

roberth committed:
build-support/rust: remove unused runCommandNoCC

siraben committed:
Merge pull request #134219 from oxalica/fcitx-vim

vimPlugins.fcitx-vim: init at 2021-08-15 and handle python3Dependencies for vim

siraben committed:
Merge pull request #134220 from r-ryantm/auto-update/oneshot

oneshot: 1.4.1 -> 1.5.0

siraben committed:
Merge pull request #134218 from r-ryantm/auto-update/onedrive

onedrive: 2.4.12 -> 2.4.13

siraben committed:
Merge pull request #134209 from SuperSandro2000/wezterm

wezterm: 20210502-154244-3f7122cb -> 20210814-124438-54e29167

siraben committed:
Merge pull request #134208 from Kranzes/auto-update/vultr-cli

vultr-cli: 2.4.1 -> 2.7.0

siraben committed:
Merge pull request #134217 from r-ryantm/auto-update/oneDNN

oneDNN: 2.2.1 -> 2.3.2

siraben committed:
Merge pull request #133883 from r-ryantm/auto-update/you-get

you-get: 0.4.1500 -> 0.4.1536

siraben committed:
Merge pull request #134211 from Kranzes/auto-update/mtools

mtools: 4.0.34 -> 4.0.35

siraben committed:
Merge pull request #134214 from gebner/sumneko236

sumneko-lua-language-server: 2.2.3 -> 2.3.6

roberth committed:
listadmin: fix deprecation

roberth committed:
nixpkgs-basic-release-checks: Check without aliases

This allows the aliases file to produce deprecation warnings
without triggering the check against warnings. This allows
for a better user experience when deprecations occur.

Moreover, Nixpkgs itself shouldn't rely on its backcompat aliases.

siraben committed:
Merge pull request #134225 from hercules-ci/runCommand-no-nocc

runCommandNoCC -> runCommand deprecation

roberth committed:
runCommandNoCC: fix ofborg status

The original pr was merged a bit too soon.

Removing the deprecation warning shouldn't be necessary, but for
now, we need to prioritize making ofborg green.

r-ryantm committed:
openfpgaloader: 0.2.6 -> 0.5.0

roberth committed:
Merge pull request #134230 from hercules-ci/remove-warning-for-ofborg-status

runCommandNoCC: fix ofborg status

Kranzes committed:
python38Packages.pyfaidx: 0.5.9.5 -> 0.6.1

AndersonTorres committed:
Merge pull request #134063 from AndersonTorres/new-cbftp

cbftp: init at 1173

veprbl committed:
Merge pull request #133350 from rasendubi/fetchzip-downloadname

fetchzip: add extension parameter

Kranzes committed:
python38Packages.numcodecs: 0.8.0 -> 0.8.1

r-ryantm committed:
osm2pgsql: 1.5.0 -> 1.5.1

etu committed:
Merge pull request #134203 from Kranzes/auto-update/php-couchbase

php74Extensions.couchbase: 3.1.2 -> 3.2.0

risicle committed:
Merge pull request #134250 from Kranzes/auto-update/python3.8-numcodecs

python38Packages.numcodecs: 0.8.0 -> 0.8.1

risicle committed:
Merge pull request #134243 from Kranzes/auto-update/python3.8-pyfaidx

python38Packages.pyfaidx: 0.5.9.5 -> 0.6.1

7c6f434c committed:
Merge pull request #134081 from kini/acl2-update

acl2: 8.3 -> 8.4, sbcl: backport ARM segfault fix

risicle committed:
Merge pull request #134232 from r-ryantm/auto-update/openfpgaloader

openfpgaloader: 0.2.6 -> 0.5.0

flokli committed:
Merge pull request #134204 from r-ryantm/auto-update/odpic

odpic: 4.1.0 -> 4.2.1

risicle committed:
Merge pull request #134256 from r-ryantm/auto-update/osm2pgsql

osm2pgsql: 1.5.0 -> 1.5.1

risicle committed:
Merge pull request #134138 from r-ryantm/auto-update/python3.8-gensim

python38Packages.gensim: 4.0.0 -> 4.0.1

risicle committed:
Merge pull request #134141 from marsam/fix-catdoc-darwin

catdoc: fix build on darwin

risicle committed:
Merge pull request #134198 from r-ryantm/auto-update/nzbget

nzbget: 21.0 -> 21.1

risicle committed:
Merge pull request #134127 from r-ryantm/auto-update/lethe

lethe: 0.5.1 -> 0.6.0

risicle committed:
Merge pull request #134114 from r-ryantm/auto-update/python3.8-mautrix

python38Packages.mautrix: 0.10.2 -> 0.10.3

Ma27 committed:
Merge pull request #134205 from TredwellGit/linux

Kernels 2021-08-15

risicle committed:
Merge pull request #134040 from r-ryantm/auto-update/vips

vips: 8.10.6 -> 8.11.2

risicle committed:
Merge pull request #134023 from r-ryantm/auto-update/libxlsxwriter

libxlsxwriter: 1.1.1 -> 1.1.3

github-actions[bot] committed:
Merge master into haskell-updates

samuelrivas commented:
> If you send a PR to haskell-updates, then it should get merged into master.
>
> I think there is some sort of action that lets us automatically create a backport, so hopefully it is not too much additional work for you.

Ok, thanks. I'll look into sending this to `haskell-updates` then.

> One thing I was wondering, is there a repo for this readline package on github? If so, can you create an issue or send a PR there and link to it from a comment in the code? If you are able to send a PR, then you should also just be able to fetchpatch from that PR, which would mean that we wouldn't need to carry this patch around in nixpkgs.
>
> If there is no upstream repo, the we'll probably have to carry around this patch in nixpkgs.

I could not find any upstream repo for this package the other time, but from a bit of googling I believe [Jan Stolarek](https://github.com/jstolarek) may be the maintainer, I have just sent an email to them.

Otherwise I could also upload the source to my github and host the patch there, but that is also relatively ugly as it would end up being a sort of fork from hackage.

Should I close this PR then? I'll open another one to `haskell-updates`

samuelrivas commented:
I went ahead and created the new PR https://github.com/NixOS/nixpkgs/pull/134295.

Still trying to find a maintainer in parallel :)
